File: week09/5/prob4.cpp
// you can use includes, for example:
#include <vector>
#include <algorithm>
using namespace std;

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    int maxProfit = 0;

    // minPriceYesterday = min(A[:i-1])
    int potentialMaxProfit;
    int minPriceYesterday = 200000;
    for(size_t i=1; i<A.size(); i++) {
        // 어제까지의 최소 = min(그저께까지의 최소, 어제 가격)
        minPriceYesterday = min(minPriceYesterday, A[i-1]);
        // maxProfit 의 후보 : 오늘 판다는 가정하에 얻을 수 있는 최대 수익
        // = 오늘 가격 - 어제까지의 최소가격
        potentialMaxProfit = A[i] - minPriceYesterday;
        if (potentialMaxProfit > maxProfit) {
            maxProfit = potentialMaxProfit;
        }
    }
    
    return maxProfit;
}

File: week09/2/Prob4.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # Implement your solution here
    if not A:
        return 0

    result = 0
    min_price = A[0]  #initial value of min price
    for i in range(1, len(A)):
        if min_price > A[i]:
            min_price = A[i]
        elif A[i] - min_price > result:
            result = A[i] - min_price
    return result


    

